app/components/SelectAreaLayer.tsx

```diff
@@ -0,0 +1,167 @@
+
+  const sourceId = `select-layer-source-${id}`
+  const fillLayerName = `select-layer-fill-${id}`;
+
+  useEffect(() => {
+    let hoverId: string | number | undefined;
+
+    if (map) {
+      const onMouseMove = (e: MapMouseEvent) => {
+        if (e.features && e.features.length > 0) {
+          const feature = e.features.at(-1);
+          const { lat, lng } = e.lngLat;
+          const aoiName = getAoiName(nameKeys, feature!.properties);
+          setHoverInfo({
+            lat,
+            lng,
+            name: aoiName
+          });
+
+          if (hoverId !== undefined) {
+            map.setFeatureState(
+              { source: sourceId, sourceLayer, id: hoverId },
+              { hover: false }
+            );
+          }
+          hoverId = feature!.id;
+          map.setFeatureState(
+            { source: sourceId, sourceLayer, id: hoverId },
+            { hover: true }
+          );
+        }
+      }
+
+      const onMouseLeave = () => {
+        setHoverInfo(undefined);
+        if (hoverId !== undefined) {
+            map.setFeatureState(
+              { source: sourceId, sourceLayer, id: hoverId },
+              { hover: false }
+            );
+        }
+        hoverId = undefined;
+      }
+
+      const onClick = (e: MapMouseEvent) => {
+        if (e.features && e.features.length > 0) {
+          // Depending on the layer, the name property has a different key.
+          // Using nameKeys of the layer config to find the right value.
+          const aoiName = getAoiName(nameKeys, e.features.at(-1)!.properties);
+
+          const selectedId = e.features.at(-1)!.id;
+          setSelectedArea(selectedId);
+          map.setFeatureState(
+            { source: sourceId, sourceLayer, id: selectedId },
+            { selected: true }
+          );
+          addContext({
+            contextType: "area",
+            content: aoiName,
+          });
+
+          map.off("mousemove", fillLayerName, onMouseMove);
+          map.off("mouseleave", fillLayerName, onMouseLeave);
+          map.off("click", fillLayerName, onClick);
+        }
+      }
+
+      map.on("mousemove", fillLayerName, onMouseMove);
+      map.on("mouseleave", fillLayerName, onMouseLeave);
+      map.on("click", fillLayerName, onClick);
+
+      return () => {
+        map.off("mousemove", fillLayerName, onMouseMove);
+        map.off("mouseleave", fillLayerName, onMouseLeave);
+        map.off("click", fillLayerName, onClick);
+      }
+    }
+  }, [map, fillLayerName, sourceId, sourceLayer, nameKeys, addContext]);
+
+  return (
+    <>
+      <Source
+        id={sourceId}
+        key={sourceId}
+        type="vector"
+        tiles={[url]}
+        promoteId="gfw_fid"
+      >
+        <Layer
+          id={fillLayerName}
+          type="fill"
+          source-layer={sourceLayer}
+          paint={{
+            'fill-color': '#fff',
+            'fill-opacity': 0
```

Comment:
```suggestion
            'fill-opacity': [
              'case',
              ['boolean', ['feature-state', 'hover'], false],
              0.16,
              ['boolean', ['feature-state', 'selected'], false],
              0.08,
              0,
            ],
```
Add a conditional opacity on fill to further indicate hover/selected areas.